src/Parachain.sol

```diff
@@ -190,37 +208,55 @@ abstract contract Parachain {
+                )
 
-        // swap bytes
-        v = ((v & 0xFF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00) >> 8)
-            | ((v & 0x00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF00FF) << 8);
+            // swap 2-byte long pairs
+            v :=
+                or(
+                    shr(16, and(v, 0xFFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000)),
+                    shl(16, and(v, 0x0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF))
+                )
 
-        // swap 2-byte long pairs
-        v = ((v & 0xFFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000) >> 16)
-            | ((v & 0x0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF0000FFFF) << 16);
+            // swap 4-byte long pairs
+            v :=
+                or(
+                    shr(32, and(v, 0xFFFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000)),
+                    shl(32, and(v, 0x00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF))
+                )
 
-        // swap 4-byte long pairs
-        v = ((v & 0xFFFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000) >> 32)
-            | ((v & 0x00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF) << 32);
+            // swap 8-byte long pairs
+            v :=
+                or(
+                    shr(64, and(v, 0xFFFFFFFFFFFFFFFF0000000000000000FFFFFFFFFFFFFFFF0000000000000000)),
+                    shl(64, and(v, 0x0000000000000000FFFFFFFFFFFFFFFF0000000000000000FFFFFFFFFFFFFFFF))
+                )
 
-        // swap 8-byte long pairs
-        v = ((v & 0xFFFFFFFFFFFFFFFF0000000000000000FFFFFFFFFFFFFFFF0000000000000000) >> 64)
-            | ((v & 0x0000000000000000FFFFFFFFFFFFFFFF0000000000000000FFFFFFFFFFFFFFFF) << 64);
-
-        // swap 16-byte long pairs
-        v = (v >> 128) | (v << 128);
+            // swap 16-byte long pairs
+            v := or(shr(128, v), shl(128, v))
+        }
     }
 
-    function registryAddress() public view returns (address) {
-        return address(registry);
+    function registryAddress() public view returns (address _addr) {
+        // return address(registry);
+        assembly {
+            _addr := sload(registry.slot)
+        }
     }
 
     /// @dev Converts provided weight to fee for XCM execution.
     /// @param overallWeight uint256 Combined weight of consumer chain's dispatchable function (wrapped in transact) and XCM instructions.
     /// @param weightToFee uint256 Fee per weight (constant multiplier)
-    function convertWeightToFee(uint256 overallWeight, uint256 weightToFee) internal pure returns (uint256) {
+    function convertWeightToFee(uint256 overallWeight, uint256 weightToFee) internal pure returns (uint256 z) {
         // overall xcm fee cost based on constantMultiplier
-        return overallWeight * weightToFee;
+        assembly {
+            z := mul(overallWeight, weightToFee)
```

Comment:
Possible to use a more explicit name like you have `_addr` in `registryAddress` instead of `z` single character variable?